File: frontend/gamingDate/lib/model/onlinestatus.dart
import 'package:gamingDate/model/onlineStatusEnum.dart';

class OnlineStatus {
  String _onlineStatusId;
  OnlineStatusEnum _onlineStatus;

  OnlineStatus(this._onlineStatusId, this._onlineStatus);

  OnlineStatusEnum get onlineStatus => _onlineStatus;

  String get onlineStatusId => _onlineStatusId;
}

File: backend/gamingdate/src/main/java/com/gamingdate/model/OnlineStatus.java
package com.gamingdate.model;

import javax.persistence.*;
import javax.validation.constraints.NotBlank;

import lombok.Data;


@Entity
@Table(name="onlinestatus")
@Data
public class OnlineStatus {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name="onlinestatusid")
    private Long onlineStatusId;

    @Column(name="status")
    @NotBlank(message = "status is mandatory")
    private String status;

    // @OneToOne(mappedBy = "onlineStatus")
    // @NotNull
    // private Profile profile;

    // default connstructor which is only used for JPA
    protected OnlineStatus(){}
    public OnlineStatus(String status){
        this.status = status;
    }
}
